src/lib/blit.c

```diff
@@ -269,6 +269,104 @@ quadrant_solver(uint32_t tl, uint32_t tr, uint32_t bl, uint32_t br,
   return egc;
 }
 
+// quadrant check for transparency. returns an EGC if we found transparent
+// quads and have solved for colors (this EGC ought then be loaded into the
+// cell). returns NULL otherwise. transparency trumps everything else in terms
+// of priority -- if even one quadrant is transparent, we will have a
+// transparent background, and lerp the rest together for foreground. we thus
+// have a 16-way conditional tree in which each EGC must show up exactly once.
+static inline const char*
+qtrans_check(cell* c, bool bgr, bool blendcolors,
+             const unsigned char* rgbbase_tl, const unsigned char* rgbbase_tr,
+             const unsigned char* rgbbase_bl, const unsigned char* rgbbase_br){
+  const int rpos = bgr ? 2 : 0;
+  const int bpos = bgr ? 0 : 2;
+  uint32_t tl = 0, tr = 0, bl = 0, br = 0;
+  channel_set_rgb(&tl, rgbbase_tl[rpos], rgbbase_tl[1], rgbbase_tl[bpos]);
+  channel_set_rgb(&tr, rgbbase_tr[rpos], rgbbase_tr[1], rgbbase_tr[bpos]);
+  channel_set_rgb(&bl, rgbbase_bl[rpos], rgbbase_bl[1], rgbbase_bl[bpos]);
+  channel_set_rgb(&br, rgbbase_br[rpos], rgbbase_br[1], rgbbase_br[bpos]);
+  const char* egc = NULL;
+  if(ffmpeg_trans_p(bgr, rgbbase_tl[3])){
```

Comment:
Actually, this entire 15-way, 4-deep conditional tree could be rewritten as an O(1) lookup table, where the index is the bit value of the transparency string, oooooh.